### DISCUSSION TITLE
Option to switch to radio-mode at end of playlist ?

### DISCUSSION THREAD

**Discussion by DutchJaFO** [music-assistant/support#813] [Feature requests and ideas]
Spotify has this nice feature where it will switch to 'radio mode' at the end of a playlist. This is an option that is set at the system level for Spotify.

Given that we now have 'radio mode' in MA as well ... this would make it even better.

**Comment by marcelveldt** [comment]
Yeah, we intend to implement something like "Don't stop the music" where you can set a preference of what to do when to queue reaches the end, such as automatically start a radio based on played tracks etc.

**Comment by DutchJaFO** [comment]
Being able to set an end-time for the playlist with an optional use as the radio to extend shorter playlists ... would probably be useful as well, because you may not always want an 'endless' playlist.

**Reply by OzGav** [reply]
This idea has been added to the backlog for a later release

**Comment by Auram** [comment]
Any updates on this?

**Reply by OzGav** [reply]
The radio mode behaviour is being looked at. No ETA can be given.

**Reply by OzGav** [reply]
This functionality is being worked on now. Hopefully will come out in the next beta or two.

**Reply by OzGav** [reply]
Don't stop the music has been implemented in beta 30

**Comment by jazzyisj** [comment]
@OzGav  Is there documentation on how to use this feature yet?  I've looked around the UI and documentation but can't find anything.

**Reply by OzGav** [reply]
No and we have found a bug in it so that is being discussed. Essentially it is either on or off. Currently, it is defaulted to on. If you are finding it isn’t always (or ever) working then that is the bug.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
